# Importation

In [1]:
# numpy and pandas for data manipulation
import numpy as np
import pandas as pd 

# sklearn preprocessing for dealing with categorical variables
from sklearn.preprocessing import LabelEncoder

# File system manangement
import os

# Suppress warnings 
import warnings
warnings.filterwarnings('ignore')

# matplotlib and seaborn for plotting
import matplotlib.pyplot as plt
import seaborn as sns

In [49]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', None)

# Création des dataframes

In [5]:
import os
import pandas as pd

def read_csv_files_in_directory(directory_path, encoding='utf-8'):
    csv_files = [f for f in os.listdir(directory_path) if f.endswith('.csv')]
    dataframes = {}
    
    for file in csv_files:
        file_path = os.path.join(directory_path, file)
        df_name = f"df_{os.path.splitext(file)[0]}"
        try:
            dataframes[df_name] = pd.read_csv(file_path, encoding=encoding)
            print(f"DataFrame '{df_name}' created with shape {dataframes[df_name].shape}")
        except UnicodeDecodeError:
            print(f"UnicodeDecodeError for file {file}. Trying 'ISO-8859-1' encoding...")
            dataframes[df_name] = pd.read_csv(file_path, encoding='ISO-8859-1')
            print(f"DataFrame '{df_name}' created with shape {dataframes[df_name].shape} using 'ISO-8859-1' encoding.")
    
    return dataframes


In [6]:
directory = r'C:\Users\mauge\Openclassrooms\Projet 7 - Implémenter un modèle de scoring\Projet+Mise+en+prod+-+home-credit-default-risk'
dataframes = read_csv_files_in_directory(directory)

DataFrame 'df_application_test' created with shape (48744, 121)
DataFrame 'df_application_train' created with shape (307511, 122)
DataFrame 'df_bureau' created with shape (1716428, 17)
DataFrame 'df_bureau_balance' created with shape (27299925, 3)
DataFrame 'df_credit_card_balance' created with shape (3840312, 23)
UnicodeDecodeError for file HomeCredit_columns_description.csv. Trying 'ISO-8859-1' encoding...
DataFrame 'df_HomeCredit_columns_description' created with shape (219, 5) using 'ISO-8859-1' encoding.
DataFrame 'df_installments_payments' created with shape (13605401, 8)
DataFrame 'df_POS_CASH_balance' created with shape (10001358, 8)
DataFrame 'df_previous_application' created with shape (1670214, 37)
DataFrame 'df_sample_submission' created with shape (48744, 2)


In [13]:
# Réaffecter les DataFrames à des variables globales
for df_name, df in dataframes.items():
    globals()[df_name] = df

# Description des dataframes

In [22]:
df_application_test.head()

SK_ID_CURR NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  \
0      100001         Cash loans           F            N               Y   
1      100005         Cash loans           M            N               Y   
2      100013         Cash loans           M            Y               Y   
3      100028         Cash loans           F            N               Y   
4      100038         Cash loans           M            Y               N   

   CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  AMT_GOODS_PRICE  \
0             0          135000.0    568800.0      20560.5         450000.0   
1             0           99000.0    222768.0      17370.0         180000.0   
2             0          202500.0    663264.0      69777.0         630000.0   
3             2          315000.0   1575000.0      49018.5        1575000.0   
4             1          180000.0    625500.0      32067.0         625500.0   

  NAME_TYPE_SUITE NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
0   Unaccompanied          Working               Higher education   
1   Unaccompanied          Working  Secondary / secondary special   
2             NaN          Working               Higher education   
3   Unaccompanied          Working  Secondary / secondary special   
4   Unaccompanied          Working  Secondary / secondary special   

  NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  REGION_POPULATION_RELATIVE  \
0            Married  House / apartment                    0.018850   
1            Married  House / apartment                    0.035792   
2            Married  House / apartment                    0.019101   
3            Married  House / apartment                    0.026392   
4            Married  House / apartment                    0.010032   

   DAYS_BIRTH  DAYS_EMPLOYED  DAYS_REGISTRATION  DAYS_ID_PUBLISH  OWN_CAR_AGE  \
0      -19241          -2329            -5170.0             -812          NaN   
1      -18064          -4469            -9118.0            -1623          NaN   
2      -20038          -4458            -2175.0            -3503          5.0   
3      -13976          -1866            -2000.0            -4208          NaN   
4      -13040          -2191            -4000.0            -4262         16.0   

   FLAG_MOBIL  FLAG_EMP_PHONE  FLAG_WORK_PHONE  FLAG_CONT_MOBILE  FLAG_PHONE  \
0           1               1                0                 1           0   
1           1               1                0                 1           0   
2           1               1                0                 1           0   
3           1               1                0                 1           1   
4           1               1                1                 1           0   

   FLAG_EMAIL     OCCUPATION_TYPE  CNT_FAM_MEMBERS  REGION_RATING_CLIENT  \
0           1                 NaN              2.0                     2   
1           0  Low-skill Laborers              2.0                     2   
2           0             Drivers              2.0                     2   
3           0         Sales staff              4.0                     2   
4           0                 NaN              3.0                     2   

   REGION_RATING_CLIENT_W_CITY WEEKDAY_APPR_PROCESS_START  \
0                            2                    TUESDAY   
1                            2                     FRIDAY   
2                            2                     MONDAY   
3                            2                  WEDNESDAY   
4                            2                     FRIDAY   

   HOUR_APPR_PROCESS_START  REG_REGION_NOT_LIVE_REGION  \
0                       18                           0   
1                        9                           0   
2                       14                           0   
3                       11                           0   
4                        5                           0   

   REG_REGION_NOT_WORK_REGION  LIVE_REGION_NOT_WORK_REGION  \
0                   

In [15]:
df_application_test.shape

(48744, 121)

In [19]:
for df_name, df in dataframes.items():
    print(df_name)
    print(df.shape)

df_application_test
(48744, 121)
df_application_train
(307511, 122)
df_bureau
(1716428, 17)
df_bureau_balance
(27299925, 3)
df_credit_card_balance
(3840312, 23)
df_HomeCredit_columns_description
(219, 5)
df_installments_payments
(13605401, 8)
df_POS_CASH_balance
(10001358, 8)
df_previous_application
(1670214, 37)
df_sample_submission
(48744, 2)


In [44]:
df_HomeCredit_columns_description

Unnamed: 0                         Table                    Row  \
0             1  application_{train|test}.csv             SK_ID_CURR   
1             2  application_{train|test}.csv                 TARGET   
2             5  application_{train|test}.csv     NAME_CONTRACT_TYPE   
3             6  application_{train|test}.csv            CODE_GENDER   
4             7  application_{train|test}.csv           FLAG_OWN_CAR   
..          ...                           ...                    ...   
214         217     installments_payments.csv  NUM_INSTALMENT_NUMBER   
215         218     installments_payments.csv        DAYS_INSTALMENT   
216         219     installments_payments.csv     DAYS_ENTRY_PAYMENT   
217         220     installments_payments.csv         AMT_INSTALMENT   
218         221     installments_payments.csv            AMT_PAYMENT   

                                           Description  \
0                             ID of loan in our sample   
1    Target variable (1 - client with payment diffi...   
2          Identification if loan is cash or revolving   
3                                 Gender of the client   
4                        Flag if the client owns a car   
..                                                 ...   
214            On which installment we observe payment   
215  When the installment of previous credit was su...   
216  When was the installments of previous credit p...   
217  What was the prescribed installment amount of ...   
218  What the client actually paid on previous cred...   

                                   Special  
0                                      NaN  
1                                      NaN  
2                                      NaN  
3                                      NaN  
4                                      NaN  
..                                     ...  
214                                    NaN  
215  time only relative to the application  
216  time only relative to the application  
217                                    NaN  
218                                    NaN  

[219 rows x 5 columns]

In [23]:
df_application_train.head()

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0      100002       1         Cash loans           M            N   
1      100003       0         Cash loans           F            N   
2      100004       0    Revolving loans           M            Y   
3      100006       0         Cash loans           F            N   
4      100007       0         Cash loans           M            N   

  FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0               Y             0          202500.0    406597.5      24700.5   
1               N             0          270000.0   1293502.5      35698.5   
2               Y             0           67500.0    135000.0       6750.0   
3               Y             0          135000.0    312682.5      29686.5   
4               Y             0          121500.0    513000.0      21865.5   

   AMT_GOODS_PRICE NAME_TYPE_SUITE NAME_INCOME_TYPE  \
0         351000.0   Unaccompanied          Working   
1        1129500.0          Family    State servant   
2         135000.0   Unaccompanied          Working   
3         297000.0   Unaccompanied          Working   
4         513000.0   Unaccompanied          Working   

             NAME_EDUCATION_TYPE    NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  \
0  Secondary / secondary special  Single / not married  House / apartment   
1               Higher education               Married  House / apartment   
2  Secondary / secondary special  Single / not married  House / apartment   
3  Secondary / secondary special        Civil marriage  House / apartment   
4  Secondary / secondary special  Single / not married  House / apartment   

   REGION_POPULATION_RELATIVE  DAYS_BIRTH  DAYS_EMPLOYED  DAYS_REGISTRATION  \
0                    0.018801       -9461           -637            -3648.0   
1                    0.003541      -16765          -1188            -1186.0   
2                    0.010032      -19046           -225            -4260.0   
3                    0.008019      -19005          -3039            -9833.0   
4                    0.028663      -19932          -3038            -4311.0   

   DAYS_ID_PUBLISH  OWN_CAR_AGE  FLAG_MOBIL  FLAG_EMP_PHONE  FLAG_WORK_PHONE  \
0            -2120          NaN           1               1                0   
1             -291          NaN           1               1                0   
2            -2531         26.0           1               1                1   
3            -2437          NaN           1               1                0   
4            -3458          NaN           1               1                0   

   FLAG_CONT_MOBILE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  CNT_FAM_MEMBERS  \
0                 1           1           0        Laborers              1.0   
1                 1           1           0      Core staff              2.0   
2                 1           1           0        Laborers              1.0   
3                 1           0           0        Laborers              2.0   
4                 1           0           0      Core staff              1.0   

   REGION_RATING_CLIENT  REGION_RATING_CLIENT_W_CITY  \
0                     2                            2   
1                     1                            1   
2                     2                            2   
3                     2                            2   
4                     2                            2   

  WEEKDAY_APPR_PROCESS_START  HOUR_APPR_PROCESS_START  \
0                  WEDNESDAY                       10   
1                     MONDAY                       11   
2                     MONDAY                        9   
3                  WEDNESDAY                       17   
4                   THURSDAY                       11   

   REG_REGION_NOT_LIVE_REGION  REG_REGION_NOT_WORK_REGION  \
0                           0                           0   
1                           0                           0   
2                        

In [26]:
df_application_train.isna().mean().tolist()

[0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 3.9022994299390916e-05,
 0.0009040327012692228,
 0.004201475719567756,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.6599081008484249,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.31345545362604915,
 6.503832383231819e-06,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.5638107254699832,
 0.0021462646864665006,
 0.19825307062186392,
 0.5074972927797705,
 0.5851595552679416,
 0.48781019215572774,
 0.6649778381911542,
 0.6987229725115525,
 0.532959796560123,
 0.503487680115508,
 0.4976082156410665,
 0.6784862980511266,
 0.5937673774271489,
 0.6835495315614726,
 0.5019332641759157,
 0.6943296337366793,
 0.5517916432257708,
 0.5074972927797705,
 0.5851595552679416,
 0.48781019215572774,
 0.6649778381911542,
 0.6987229725115525,
 0.532959796560123,
 0.503487680115508,
 0.4976082156410665,
 0.6784862980511266,
 0.5937673774271489,
 0.6835495315614726,
 0.5019332641759157,
 0.6943296337366793,
 0.5517916432257708

## df_bureau

In [27]:
df_bureau.head()

SK_ID_CURR  SK_ID_BUREAU CREDIT_ACTIVE CREDIT_CURRENCY  DAYS_CREDIT  \
0      215354       5714462        Closed      currency 1         -497   
1      215354       5714463        Active      currency 1         -208   
2      215354       5714464        Active      currency 1         -203   
3      215354       5714465        Active      currency 1         -203   
4      215354       5714466        Active      currency 1         -629   

   CREDIT_DAY_OVERDUE  DAYS_CREDIT_ENDDATE  DAYS_ENDDATE_FACT  \
0                   0               -153.0             -153.0   
1                   0               1075.0                NaN   
2                   0                528.0                NaN   
3                   0                  NaN                NaN   
4                   0               1197.0                NaN   

   AMT_CREDIT_MAX_OVERDUE  CNT_CREDIT_PROLONG  AMT_CREDIT_SUM  \
0                     NaN                   0         91323.0   
1                     NaN                   0        225000.0   
2                     NaN                   0        464323.5   
3                     NaN                   0         90000.0   
4                 77674.5                   0       2700000.0   

   AMT_CREDIT_SUM_DEBT  AMT_CREDIT_SUM_LIMIT  AMT_CREDIT_SUM_OVERDUE  \
0                  0.0                   NaN                     0.0   
1             171342.0                   NaN                     0.0   
2                  NaN                   NaN                     0.0   
3                  NaN                   NaN                     0.0   
4                  NaN                   NaN                     0.0   

       CREDIT_TYPE  DAYS_CREDIT_UPDATE  AMT_ANNUITY  
0  Consumer credit                -131          NaN  
1      Credit card                 -20          NaN  
2  Consumer credit                 -16          NaN  
3      Credit card                 -16          NaN  
4  Consumer credit                 -21          NaN

**DAYS_CREDIT** mesure le temps écoulé, en jours, entre la date à laquelle le client a fait une demande de crédit enregistrée par le bureau de crédit et la date de la demande de crédit actuelle.  
**CREDIT_DAY_OVERDUE** mesure le nombre de jours de retard accumulés par le client sur ce crédit spécifique au moment où il a fait la demande de crédit pour le prêt lié dans l'échantillon de données.

In [52]:
df_HomeCredit_columns_description[df_HomeCredit_columns_description['Table']=='bureau.csv']

Unnamed: 0       Table                     Row  \
122         125  bureau.csv              SK_ID_CURR   
123         126  bureau.csv            SK_BUREAU_ID   
124         127  bureau.csv           CREDIT_ACTIVE   
125         128  bureau.csv         CREDIT_CURRENCY   
126         129  bureau.csv             DAYS_CREDIT   
127         130  bureau.csv      CREDIT_DAY_OVERDUE   
128         131  bureau.csv     DAYS_CREDIT_ENDDATE   
129         132  bureau.csv       DAYS_ENDDATE_FACT   
130         133  bureau.csv  AMT_CREDIT_MAX_OVERDUE   
131         134  bureau.csv      CNT_CREDIT_PROLONG   
132         135  bureau.csv          AMT_CREDIT_SUM   
133         136  bureau.csv     AMT_CREDIT_SUM_DEBT   
134         137  bureau.csv    AMT_CREDIT_SUM_LIMIT   
135         138  bureau.csv  AMT_CREDIT_SUM_OVERDUE   
136         139  bureau.csv             CREDIT_TYPE   
137         140  bureau.csv      DAYS_CREDIT_UPDATE   
138         141  bureau.csv             AMT_ANNUITY   

                                                                                                             Description  \
122  ID of loan in our sample - one loan in our sample can have 0,1,2 or more related previous credits in credit bureau    
123            Recoded ID of previous Credit Bureau credit related to our loan (unique coding for each loan application)   
124                                                                    Status of the Credit Bureau (CB) reported credits   
125                                                                         Recoded currency of the Credit Bureau credit   
126                                   How many days before current application did client apply for Credit Bureau credit   
127                       Number of days past due on CB credit at the time of application for related loan in our sample   
128                                  Remaining duration of CB credit (in days) at the time of application in Home Credit   
129                        Days since CB credit ended at the time of application in Home Credit (only for closed credit)   
130                Maximal amount overdue on the Credit Bureau credit so far (at application date of loan in our sample)   
131                                                                How many times was the Credit Bureau credit prolonged   
132                                                                   Current credit amount for the Credit Bureau credit   
133                                                                                 Current debt on Credit Bureau credit   
134                                                        Current credit limit of credit card reported in Credit Bureau   
135                                                                       Current amount overdue on Credit Bureau credit   
136                                                                         Type of Credit Bureau credit (Car, cash,...)   
137                       How many days before loan application did last information about the Credit Bureau credit come   
138                                                                                  Annuity of the Credit Bureau credit   

                                   Special  
122                                 hashed  
123                                 hashed  
124                                    NaN  
125                                recoded  
126  time only relative to the application  
127                                    NaN  
128  time only relative to the application  
129  time only relative to the application  
130                                    NaN  
131                                    NaN  
132                                    NaN  
133                                    NaN  
134                                    NaN  
135                                    NaN  
136                                    NaN  
137  time only relative to the application  
138             

## df_bureau_balance

In [28]:
df_bureau_balance.head()

SK_ID_BUREAU  MONTHS_BALANCE STATUS
0       5715448               0      C
1       5715448              -1      C
2       5715448              -2      C
3       5715448              -3      C
4       5715448              -4      C

In [38]:
df_bureau_balance[df_bureau_balance['SK_ID_BUREAU']==5714466]

Empty DataFrame
Columns: [SK_ID_BUREAU, MONTHS_BALANCE, STATUS]
Index: []

In [57]:
unique_values_df1 = set(df_bureau['SK_ID_BUREAU'].unique())
unique_values_df2 = set(df_bureau_balance['SK_ID_BUREAU'].unique())

# Trouver les valeurs uniques communes
common_unique_values = unique_values_df1.intersection(unique_values_df2)

# Compter le nombre de valeurs uniques communes
num_common_unique_values = len(common_unique_values)

print(f"Nombre de valeurs uniques communes dans 'SK_ID_BUREAU' entre les deux DataFrames : {num_common_unique_values}")
print(f"Valeur uniques de 'SK_ID_BUREAU' dans df_bureau: {df_bureau['SK_ID_BUREAU'].nunique()}")
print(f"Valeur uniques de 'SK_ID_BUREAU' dans df_bureau_balance: {df_bureau_balance['SK_ID_BUREAU'].nunique()}")

Nombre de valeurs uniques communes dans 'SK_ID_BUREAU' entre les deux DataFrames : 774354
Valeur uniques de 'SK_ID_BUREAU' dans df_bureau: 1716428
Valeur uniques de 'SK_ID_BUREAU' dans df_bureau_balance: 817395


In [58]:
df_bureau.shape

(1716428, 17)

In [50]:
df_HomeCredit_columns_description[df_HomeCredit_columns_description['Table']=='bureau_balance.csv']

Unnamed: 0               Table             Row  \
139         142  bureau_balance.csv    SK_BUREAU_ID   
140         143  bureau_balance.csv  MONTHS_BALANCE   
141         144  bureau_balance.csv          STATUS   

                                                                                                                                                                                                                                          Description  \
139                                                                                                                                Recoded ID of Credit Bureau credit (unique coding for each application) - use this to join to CREDIT_BUREAU table    
140                                                                                                                                                                Month of balance relative to application date (-1 means the freshest balance date)   
141  Status of Credit Bureau loan during the month (active, closed, DPD0-30,
 [C means closed, X means status unknown, 0 means no DPD, 1 means maximal did during month between 1-30, 2 means DPD 31-60,
 5 means DPD 120+ or sold or written off ] )   

                                   Special  
139                                 hashed  
140  time only relative to the application  
141                                    NaN

### Explication des Statuts de Prêt dans la Colonne "Status of Credit Bureau Loan"
#### Valeurs possibles et leurs significations :

1. **C** : **Closed (Clos)**
   - Ce statut indique que le prêt a été clôturé, signifiant généralement que le prêt a été entièrement remboursé et que l'emprunteur n'a plus de dettes envers le prêteur pour ce prêt spécifique.

2. **X** : **Status Unknown (Statut Inconnu)**
   - Ce statut signifie que le statut du prêt pour ce mois est inconnu ou n'a pas été rapporté. Il pourrait y avoir des raisons administratives pour lesquelles le statut n'a pas été enregistré, ou cela pourrait être une erreur de reporting.

3. **0** : **No DPD (Pas de retard)**
   - Cette valeur signifie qu'il n'y a eu aucun retard de paiement durant le mois. L'emprunteur a respecté toutes les obligations de paiement pour ce mois.

4. **1** : **DPD 1-30 (Retard de 1 à 30 jours)**
   - Cette valeur indique que le paiement mensuel était en retard de 1 à 30 jours durant le mois. C'est un indicateur que l'emprunteur a commencé à avoir des difficultés à rembourser son prêt à temps, mais qu'il n'est pas encore en grave défaut de paiement.

5. **2** : **DPD 31-60 (Retard de 31 à 60 jours)**
   - Cette valeur indique que le paiement mensuel était en retard de 31 à 60 jours. C'est une situation plus préoccupante, car cela montre que l'emprunteur a manqué plusieurs paiements ou qu'il est en retard depuis plus d'un mois.

6. **3, 4, 5, etc.** : **DPD prolongé (Retard prolongé)**
   - Les valeurs augmentent pour indiquer des retards plus longs :
     - **3** : Retard de 61 à 90 jours.
     - **4** : Retard de 91 à 120 jours.
     - **5** : Retard de 120+ jours, prêt vendu ou radié.
   - Un retard de 120 jours ou plus (valeur 5) est extrêmement grave et indique que le prêt pourrait avoir été vendu à une agence de recouvrement ou radié comme une perte par la banque.

#### Contexte :

Ces types de données sont couramment utilisées par les bureaux de crédit pour évaluer la solvabilité d'un emprunteur. Un historique de paiements réguliers (indiqué par beaucoup de `0`) est un bon signe, tandis que des DPD prolongés (par exemple, `2`, `3`, `4`, `5`) signalent des risques plus élevés et peuvent affecter négativement la cote de crédit d'un individu.

In [53]:
df_bureau_balance[df_bureau_balance['SK_ID_BUREAU']==5715448]

SK_ID_BUREAU  MONTHS_BALANCE STATUS
0        5715448               0      C
1        5715448              -1      C
2        5715448              -2      C
3        5715448              -3      C
4        5715448              -4      C
5        5715448              -5      C
6        5715448              -6      C
7        5715448              -7      C
8        5715448              -8      C
9        5715448              -9      0
10       5715448             -10      0
11       5715448             -11      X
12       5715448             -12      X
13       5715448             -13      X
14       5715448             -14      0
15       5715448             -15      0
16       5715448             -16      0
17       5715448             -17      0
18       5715448             -18      0
19       5715448             -19      0
20       5715448             -20      X
21       5715448             -21      X
22       5715448             -22      X
23       5715448             -23      X
24       5715448             -24      X
25       5715448             -25      X
26       5715448             -26      X

## df_previous_application

In [55]:
df_previous_application.head()

SK_ID_PREV  SK_ID_CURR NAME_CONTRACT_TYPE  AMT_ANNUITY  AMT_APPLICATION  \
0     2030495      271877     Consumer loans     1730.430          17145.0   
1     2802425      108129         Cash loans    25188.615         607500.0   
2     2523466      122040         Cash loans    15060.735         112500.0   
3     2819243      176158         Cash loans    47041.335         450000.0   
4     1784265      202054         Cash loans    31924.395         337500.0   

   AMT_CREDIT  AMT_DOWN_PAYMENT  AMT_GOODS_PRICE WEEKDAY_APPR_PROCESS_START  \
0     17145.0               0.0          17145.0                   SATURDAY   
1    679671.0               NaN         607500.0                   THURSDAY   
2    136444.5               NaN         112500.0                    TUESDAY   
3    470790.0               NaN         450000.0                     MONDAY   
4    404055.0               NaN         337500.0                   THURSDAY   

   HOUR_APPR_PROCESS_START FLAG_LAST_APPL_PER_CONTRACT  \
0                       15                           Y   
1                       11                           Y   
2                       11                           Y   
3                        7                           Y   
4                        9                           Y   

   NFLAG_LAST_APPL_IN_DAY  RATE_DOWN_PAYMENT  RATE_INTEREST_PRIMARY  \
0                       1                0.0               0.182832   
1                       1                NaN                    NaN   
2                       1                NaN                    NaN   
3                       1                NaN                    NaN   
4                       1                NaN                    NaN   

   RATE_INTEREST_PRIVILEGED NAME_CASH_LOAN_PURPOSE NAME_CONTRACT_STATUS  \
0                  0.867336                    XAP             Approved   
1                       NaN                    XNA             Approved   
2                       NaN                    XNA             Approved   
3                       NaN                    XNA             Approved   
4                       NaN                Repairs              Refused   

   DAYS_DECISION      NAME_PAYMENT_TYPE CODE_REJECT_REASON  NAME_TYPE_SUITE  \
0            -73  Cash through the bank                XAP              NaN   
1           -164                    XNA                XAP    Unaccompanied   
2           -301  Cash through the bank                XAP  Spouse, partner   
3           -512  Cash through the bank                XAP              NaN   
4           -781  Cash through the bank                 HC              NaN   

  NAME_CLIENT_TYPE NAME_GOODS_CATEGORY NAME_PORTFOLIO NAME_PRODUCT_TYPE  \
0         Repeater              Mobile            POS               XNA   
1         Repeater                 XNA           Cash            x-sell   
2         Repeater                 XNA           Cash            x-sell   
3         Repeater                 XNA           Cash            x-sell   
4         Repeater                 XNA           Cash           walk-in   

              CHANNEL_TYPE  SELLERPLACE_AREA NAME_SELLER_INDUSTRY  \
0             Country-wide                35         Connectivity   
1           Contact center                -1                  XNA   
2  Credit and cash offices                -1                  XNA   
3  Credit and cash offices                -1                  XNA   
4  Credit and cash offices                -1                  XNA   

   CNT_PAYMENT NAME_YIELD_GROUP       PRODUCT_COMBINATION  DAYS_FIRST_DRAWING  \
0         12.0           middle  POS mobile with interest            365243.0   
1         36.0       low_action          Cash X-Sell: low            365243.0   
2         12.0             high         Cash X-Sell: high            365243.0   
3         12.0           middle       Cash X-Sell: middle            365243.0   
4         24.0             high         Cash Street: high                 NaN   

   D

In [56]:
df_HomeCredit_columns_description[df_HomeCredit_columns_description['Table']=='previous_application.csv']

Unnamed: 0                     Table                          Row  \
173         176  previous_application.csv                  SK_ID_PREV    
174         177  previous_application.csv                   SK_ID_CURR   
175         178  previous_application.csv           NAME_CONTRACT_TYPE   
176         179  previous_application.csv                  AMT_ANNUITY   
177         180  previous_application.csv              AMT_APPLICATION   
178         181  previous_application.csv                   AMT_CREDIT   
179         182  previous_application.csv             AMT_DOWN_PAYMENT   
180         183  previous_application.csv              AMT_GOODS_PRICE   
181         184  previous_application.csv   WEEKDAY_APPR_PROCESS_START   
182         185  previous_application.csv      HOUR_APPR_PROCESS_START   
183         186  previous_application.csv  FLAG_LAST_APPL_PER_CONTRACT   
184         187  previous_application.csv       NFLAG_LAST_APPL_IN_DAY   
185         188  previous_application.csv             NFLAG_MICRO_CASH   
186         189  previous_application.csv            RATE_DOWN_PAYMENT   
187         190  previous_application.csv        RATE_INTEREST_PRIMARY   
188         191  previous_application.csv     RATE_INTEREST_PRIVILEGED   
189         192  previous_application.csv       NAME_CASH_LOAN_PURPOSE   
190         193  previous_application.csv         NAME_CONTRACT_STATUS   
191         194  previous_application.csv                DAYS_DECISION   
192         195  previous_application.csv            NAME_PAYMENT_TYPE   
193         196  previous_application.csv           CODE_REJECT_REASON   
194         197  previous_application.csv              NAME_TYPE_SUITE   
195         198  previous_application.csv             NAME_CLIENT_TYPE   
196         199  previous_application.csv          NAME_GOODS_CATEGORY   
197         200  previous_application.csv               NAME_PORTFOLIO   
198         201  previous_application.csv            NAME_PRODUCT_TYPE   
199         202  previous_application.csv                 CHANNEL_TYPE   
200         203  previous_application.csv             SELLERPLACE_AREA   
201         204  previous_application.csv         NAME_SELLER_INDUSTRY   
202         205  previous_application.csv                  CNT_PAYMENT   
203         206  previous_application.csv             NAME_YIELD_GROUP   
204         207  previous_application.csv          PRODUCT_COMBINATION   
205         208  previous_application.csv           DAYS_FIRST_DRAWING   
206         209  previous_application.csv               DAYS_FIRST_DUE   
207         210  previous_application.csv    DAYS_LAST_DUE_1ST_VERSION   
208         211  previous_application.csv                DAYS_LAST_DUE   
209         212  previous_application.csv             DAYS_TERMINATION   
210         213  previous_application.csv    NFLAG_INSURED_ON_APPROVAL   

                                                                                                                                                                                                                                                             Description  \
173                               ID of previous credit in Home credit related to loan in our sample. (One loan in our sample can have 0,1,2 or more previous loan applications in Home Credit, previous application could, but not necessarily have to lead to credit)    
174                                                                                                                                                                                                                                             ID of loan in our sample   
175                                                                                                                                                                              Contract product type (Cash loan, consumer loan [POS] ,...) of the previous application   
176                                           